File: algo-method.com/tasks/1076/main.py
# input functions --------------------------------------------------------------
def input_int_list() -> "list[int]":
    return list(map(int, input().split()))


def main() -> None:
    _N, _K = input_int_list()
    V = input_int_list()

    ans = sum(1 << v for v in V)
    print(ans)


main()

File: algo-method.com/tasks/1076/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];
const print = (...messages: any) => console.log(...messages);

const sum = (array: number[]): number => array.reduce((a, b) => a + b, 0);

// input functions -------------------------------------------------------------
const inputNumberArray = (): number[] => input().split(" ").map(Number);
// input functions -------------------------------------------------------------

const main = () => {
  const [N, K] = inputNumberArray();
  const V = inputNumberArray();

  const ans = sum(V.map((v) => 1 << v));
  print(ans);
};

main();
